core/src/test/java/org/apache/iceberg/hadoop/TestHadoopTables.java

```diff
@@ -92,15 +90,15 @@ public void testDropTableWithPurge() throws IOException {
         .isInstanceOf(NoSuchTableException.class)
         .hasMessageStartingWith("Table does not exist");
 
-    Assert.assertEquals(0, dataDir.listFiles().length);
-    Assert.assertFalse(tableDir.exists());
+    Assertions.assertThat(dataDir.listFiles()).hasSize(0);
+    Assertions.assertThat(tableDir).doesNotExist();
 
-    Assert.assertFalse(TABLES.dropTable(tableDir.toURI().toString()));
+    Assertions.assertThat(TABLES.dropTable(tableDir.toURI().toString())).isFalse();
   }
 
   @Test
-  public void testDropTableWithoutPurge() throws IOException {
-    File dataDir = temp.newFolder();
+  public void testDropTableWithoutPurge(@TempDir Path temp) throws IOException {
```

Comment:
why do we need the `@TempDir Path temp` parameter here? We should be able to use `tableDir`

Reply:
I replaced them all with tableDir.